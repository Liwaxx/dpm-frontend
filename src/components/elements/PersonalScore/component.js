import React from 'react';
import Avatar from '@material-ui/core/Avatar';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';
import Card from '../Card';
import Text from '../Text';
import { IMAGES } from '../../../configs';

export default class Component extends React.Component {
  render() {
    const { classes, data } = this.props;

    if (data) {
      return (
        <Card className={classes.card}>
          <CardContent className={classes.bungkus}>
            <Text type="display" className={classes.title} >Personal Score</Text>
            <Text type="paragraph" className={classes.nama}>Frontend Mobile Developer</Text>

            <CardContent className={classes.row}>
              <Avatar alt="Remy Sharp" src={IMAGES.USER} className={classes.Avatar} />
            </CardContent>
            <Text type="subheader" className={classes.namaOrang}>Rizky Ramadhan</Text>

            <Typography type="subheader" className={classes.angka}>
              <img src={IMAGES.STAR} className={classes.star} /> 5.0</Typography>

            <div className={classes.keterangan}>
              <Text type="paragraph">Project History</Text>
              <Text type="subheader" className={classes.warna}>Bulir</Text>
              <br />
              <Text type="paragraph">Key Points</Text>
              {/* <Text type="subheader" className={classes.warna}>Research</Text>
              <Text type="subheader" className={classes.warna}>Design </Text>
              <Text type="subheader" className={classes.warna}>Communicative</Text> */}
              <br />
              <Text type="paragraph">Need Improvement</Text>
              {/* <Text type="subheader" className={classes.warna}>Visual Design</Text>
              <Text type="subheader" className={classes.warna}>Leadership</Text>
              <Text type="subheader" className={classes.warna}>Time Management</Text> */}
            </div>
            <div>
              <Text type="paragraph" className={classes.detail}>DETAIL</Text>
            </div>
          </CardContent>
        </Card>
      );
    } else {
      return(
        <Text type="paragraph">Loading...</Text> 
      );
    }
  }
}

Component.propTypes = {
  classes: PropTypes.object.isRequired,
  text: PropTypes.string,
  url: PropTypes.string,
  data: PropTypes.object
};
