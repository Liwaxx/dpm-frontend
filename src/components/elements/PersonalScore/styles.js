const styles = {
  card: {
    width: '200px',
    height: '510px',
  },
  title: {
    paddingTop:-100,
    marginTop: '-8px',
    marginLeft: '-20px',
    marginRight: '-10px'
  },
  nama: {
    marginLeft: '-10px'
  },
  namaOrang: {
    marginLeft: '-10px'
  },
  star: {
    width: '13.5px',
    height: '13.5px',
    marginLeft: '-10px'
  },
  keterangan: {
    textAlign: 'left',
    marginLeft: '-9px'
  },
  bungkus:{
    borderLeft:'3px solid #54BAED',
    height:10,
  },
  input: {
    display: 'none',
    backroundColor: '#EBFFS'
  },
  detail: {
    color: '#54BAED',
    textAlign: 'left',
    fontWeight: 400,
    lineHeight:'17px',
    width: '47px',
    marginLeft: 80,
    marginTop: 15,
  },
  row: {
    display: 'flex',
    justifyContent: 'center',
  },
  Avatar: {
    width: 80,
    height: 80,
    marginTop: '-10px',
    marginLeft: '-10px'
  },
  warna: {
    color: '#8D98AA',
  }
};

export default styles;