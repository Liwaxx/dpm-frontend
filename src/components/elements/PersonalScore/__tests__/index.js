import React from 'react';
import renderer from 'react-test-renderer';
import PersonalScore from '../index';

jest.mock('../../Card');
jest.mock('../../Text');
jest.mock('../../../../configs');

describe('PersonalScore', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<PersonalScore />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with data', () => {
    const data = { tes: 'tes' };
    const tree = renderer
      .create(<PersonalScore data={data} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});