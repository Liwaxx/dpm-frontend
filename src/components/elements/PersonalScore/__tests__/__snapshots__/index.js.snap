// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalScore renders correctly 1`] = `null`;

exports[`PersonalScore renders with data 1`] = `null`;
